Add SecureBootTest: VariableAttributes, VariableUpdates and ImageLoadingTest for UEFI Secure Boot validation

sathishas89 opened the pull request:
SecureBootTest supports following test case:
1) VariableAttributes
   - Validates the attributes (e.g., non-volatile, authenticated write access) of critical Secure Boot variables: PK, KEK, db, dbx.

2) VariableUpdates
   - Tests runtime update paths for Secure Boot variables to ensure UEFI-enforced access control policies are in place.

3) ImageLoadingTest
   - Verifies image authentication and signature validation mechanisms during UEFI boot.

Spefication References:
-  BBSR v1.3 (Base Boot Security Requirements)
   Section 3.2: Secure Boot
   Documentation Link: https://documentation-service.arm.com/static/65e84577837c4d065f655931

-  UEFI Specification v2.11
   Section 32: Secure Boot and Driver Signing
   Documentation Link: https://uefi.org/specs/UEFI/2.11/UEFI_Spec_2_11_Dec2024.pdf

Documentation:
-  Refer to uefi-sct/Doc/TestCaseSpec/03_Services_Boot_Services.md and uefi-sct/Doc/TestCaseSpec/04_Services_Runtime_Services_Test.md
   for test case descriptions, procedures, expected behavior, and rationale.

Co-authors:
Co-authored-by: Amrathesh <Amrathesh@arm.com>
Co-authored-by: Stuart Yoder <Stuart.Yoder@arm.com>

xypron commented on uefi-sct/Doc/UserGuide/SecureBootConfigurationGuide.md:
```diff
@@ -0,0 +1,105 @@
+
+4. **Proceed with Key Enrollment**
+   - Follow platform-specific procedures to load PK, KEK, db, and dbx.
+   - Example: For Raspberry Pi, refer to the section “Enrolling Secure Boot Keys on Raspberry Pi Using UEFI Firmware.”
```
The Raspberry Pi is typically not booted via UEFI. Why don't we use EDK II as the reference in the documenation?

xypron commented on uefi-sct/Doc/UserGuide/SecureBootConfigurationGuide.md:
```diff
@@ -0,0 +1,105 @@
+
+# Enrolling and Disabling Secure Boot Keys on EDK2-Based UEFI Firmware
+
```
Why would we add this documentation to the SCT and not to EDK II?

xypron commented on uefi-sct/Doc/UserGuide/SecureBootTestKeyGenerationAndSetupGuide.md:
```diff
@@ -0,0 +1,85 @@
+### 1. Install Required Tools
+
+Run the following command to install the necessary tools:
+
```
Please, mention which tool is needed before describing how to install it on specific operating systems.

xypron commented on uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/Guid.h:
```diff
@@ -0,0 +1,236 @@
+
+  Copyright 2006 - 2016 Unified EFI, Inc.<BR>
+  Copyright (c) 2010 - 2018, Intel Corporation. All rights reserved.<BR>
+  Copyright 2021 - 2025, Arm Ltd.
```
Please, don't copy arbitrary copyright statements.

xypron commented on uefi-sct/SctPkg/TestCase/UEFI/EFI/RuntimeServices/SecureBoot/BlackBoxTest/VariableAttributesBBTest.c:
```diff
@@ -0,0 +1,474 @@
+  THE PROGRAM IS DISTRIBUTED UNDER THE BSD LICENSE ON AN "AS IS" BASIS,
+  WITHOUT WARRANTIES OR REPRESENTATIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED.
+ 
+**/
```
Please, add in each test as short description of what is tested.

sunnywang-arm reviewed (commented):
Since this is a big code change, could you give us a mapping between this change and https://github.com/ARM-software/bbr-acs/tree/main/bbsr/sct-tests/SecureBoot/BlackBoxTest? What files are newly added for edk2-test upstream?

edhay reviewed (approved):
Changes are reviewed OK.

sathishas89 commented:
> Is there an accompanying merge request for the SCT specification?

Hi @xypron, yes. The SCT specification text is included in the same PR (edk2-test#276). It updates the spec under uefi-sct/Doc/TestCaseSpec/ (both 03_Services_Boot_Services.md and 04_Services_Runtime_Services_Test.md) and also adds related user guides (SecureBootConfigurationGuide.md, SecureBootTestKeyGenerationAndSetupGuide.md). 
[GitHub](https://github.com/tianocore/edk2-test/pull/276/files)

You can see those doc changes listed in the PR’s “Files changed” view.

sathishas89 commented:
Hi @xypron , thanks for the review. I’ve pushed updates addressing all your comments.

Please take another look and let me know if anything else needs to be adjusted.

sunnywang-arm reviewed (approved):
> The following items have been newly added during the upstreaming from the BBR repository: SampleAppForSecureBootTest1 SampleAppForSecureBootTest2 SampleAppForSecureBootTest3
> 
> Additionally, the static binaries have been removed.

Thanks for clarifying, @sathishas89 The changes for adding three newly added applications look good to me. Only a minor comment that can be addressed after we merge this pull request. If everything else is the same as the ones being used in SystemReady ACS, I think we're good to go.
